src/hooks/useDatePicker.ts

```diff
@@ -0,0 +1,223 @@
+import "../date-override";
+import { isCustomCalendar } from "../utils/helpers";
+import {
+  customCalendars,
+  getCustomCalendarLocale,
+  getCustomCalendarLocales,
+} from "../custom-calendars";
+import { SupportedCalendar } from "../types";
+import { numberingSystems } from "../constants";
+
+type DatePickerOptions = {
+  date: string;
+  options: LocaleOptions;
+  onDateSelect: ({
+    calendarDate,
+    isoDate,
+  }: {
+    calendarDate: Temporal.ZonedDateTime;
+    isoDate: Temporal.ZonedDateTime;
+  }) => void;
+};
+
+type LocaleOptions = {
+  locale: string;
+  calendar: SupportedCalendar;
+  timeZone?: Temporal.TimeZoneLike | Temporal.TimeZoneProtocol;
+  numberingSystem?: string;
+  weekDayFormat?: "narrow" | "short" | "long";
+};
+
+export const useDatePicker = ({
+  onDateSelect,
+  date,
+  options,
+}: DatePickerOptions) => {
+  if (!options.calendar) {
+    throw new Error("options should include calendar");
+  }
+
+  const prevDateStringRef = useRef(date);
+
+  let calendar: Temporal.CalendarProtocol | Temporal.CalendarLike;
+
+  const { calendar: calendarFromOptions } = options;
+
+  const customCalendar = customCalendars[calendarFromOptions]?.calendar;
+
+  calendar = customCalendar || options.calendar;
+
+  const isCustom = isCustomCalendar(calendarFromOptions);
+
+  if (isCustom) {
+    const customLocalisations = getCustomCalendarLocales(calendar) || {};
+    const allowedLocales = Object.keys(customLocalisations);
+    if (!allowedLocales.includes(options.locale)) {
+      throw new Error(
+        `For the custom calendar "${
+          options.calendar
+        }", only specific locales are allowed: ${allowedLocales.join(", ")}`
+      );
+    }
+  }
+
+  const temporalCalendar = useMemo(
+    () => Temporal.Calendar.from(calendar),
+    [calendar]
+  );
+  const temporalTimeZone = useMemo(
+    () =>
+      Temporal.TimeZone.from(
+        options.timeZone ||
+          Intl?.DateTimeFormat?.().resolvedOptions?.()?.timeZone ||
+          "UTC"
+      ),
+    [options]
+  );
+
+  const todayZdt = useMemo(
+    () =>
+      Temporal.Now.zonedDateTime(temporalCalendar)
+        .withTimeZone(temporalTimeZone)
+        .startOfDay(),
+    [temporalCalendar, temporalTimeZone]
+  );
+
+  const selectedDateZdt = useMemo(
+    () =>
+      date
+        ? Temporal.PlainDate.from(date)
+            .toZonedDateTime({
+              timeZone: temporalTimeZone,
+            })
+            .withCalendar(temporalCalendar)
+        : null,
+    [date, temporalTimeZone, temporalCalendar]
+  );
+
+  const [firstZdtOfVisibleMonth, setFirstZdtOfVisibleMonth] = useState(() => {
+    const zdt = selectedDateZdt || todayZdt;
+    return zdt.with({ day: 1 });
+  });
+
+  const { locale } = options;
+
+  const localeOptions = useMemo(
+    () => ({
+      locale,
+      calendar: temporalCalendar,
+      timeZone: temporalTimeZone,
+      weekDayFormat: options.weekDayFormat || "narrow",
+      numberingSystem: options.numberingSystem,
```

Comment:
So.... As I understand it, a locale will come a default numbering system. But `options.numberingSystem` can be undefined. We pass the `localeOptions` to various other hooks and helpers. I've had a quick look, and think it should be fine, in the sense that ultimately `options.numberingSystem` will be passed to some `Temporal` or `Intl` method which will probably resolve the undefined locale correctly internally.....

I wonder if this is the best way forward though... As an alternative, we could ensure that all options are populated 
 before we proceed. So these would either have the value passed to the hook, or some resolved value. I just think it would be kind of nice to safely assume all (resolved) options are populated before we pass them to other places. We might end up adding some custom logic later on that isn't so resilient to undefined values.